app/views/sp_components/edit.html.erb

```diff
@@ -0,0 +1,38 @@
+<%= page_title t('components.edit_sp_title') %>
+
+<fieldset class="govuk-fieldset">
+  <legend class="govuk-fieldset__legend govuk-fieldset__legend--xl">
+    <h1 class="govuk-fieldset__heading"><%= t('components.edit_your_component', type: COMPONENT_TYPE::SP_SHORT) %></h1>
+  </legend>
+
+  <%= error_summary_for(@event.errors, :component) if @event %>
+  <%= form_for @component, url: sp_component_path, as: :component, method: "patch" do |f| %>
+    <div class="govuk-form-group <%= 'govuk-form-group--error' if @event && @event.errors.key?(:name) %>">
```

Comment:
You could make The O Father proud:

~if @event && @event.errors.key?(:name)~ => if @event&.errors.key?(:name)

😝

Reply:
thanks! Have amended!